components/calendar/src/astronomy.rs

```diff
@@ -0,0 +1,657 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// This file contains important structs and functions relating to location,
+/// time, and astronomy; these are intended for calender calculations and based off
+/// _Calendrical Calculations_ by Reingold & Dershowitz.
+use crate::iso::Iso;
+use crate::rata_die::RataDie;
+use crate::types::Moment;
+use crate::{Date, Gregorian};
+
+#[derive(Debug)]
+pub struct Location {
+    latitude: f32,  // latitude from -90 to 90
+    longitude: f32, // longitude from -180 to 180
+    elevation: f32, // elevation in meters
+}
+
+// The mean synodic month in days of 86400 atomic seconds
+// (86400 = 24 * 60 * 60)
+pub const MEAN_SYNODIC_MONTH: f64 = 29.530588861;
+
+// The Moment of noon on January 1, 2000
+pub const J2000: Moment = Moment::new(730120.5);
+
+impl Location {
+    /// Create a location; latitude is from -90 to 90, and longitude is from -180 to 180
+    ///
+    /// ```rust
+    /// use icu::calendar::astronomy::Location;
+    ///
+    /// let location: Location = Location::new(29.3, -132.6, 1032.5);
```

Comment:
suggestion: Since your docs mention that certain ranges are dis-/allowed, and you do indeed call `.check()`, then  we should include an example when input is out of range to show what happens. Related to the point about the design of the constructor, if the constructor can return an Error on invalid types, you could choose to assert an error when giving an out of bounds value for one of the parameters.

Reply:
Sounds good, and you added a unit test (`check_location_errors()`) to assert that outside of the happy case when trying to construction a `Location`, we get errors as we expect. That's good test coverage.